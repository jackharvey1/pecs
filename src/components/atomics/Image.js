import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const ShadowedImage = styled.img`
    box-shadow: 0 2px 8px 0 rgba(0, 0, 0, .25);

    margin: auto;

    width: 125px
    height: 125px

    @media (min-width: 1024px) {
        width: 150px;
        height: 150px;
    }
`;

const Image = ({ src, alt }) =>
    (<ShadowedImage
        src={src}
        alt={alt}
    />);

Image.propTypes = {
    src: PropTypes.string.isRequired,
    alt: PropTypes.string.isRequired,
};

export default Image;

