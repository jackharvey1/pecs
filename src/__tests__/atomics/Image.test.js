import React from 'react';
import renderer from 'react-test-renderer';
import Image from '../../components/atomics/Image';

describe('Image component', () => {
    it('renders as intended', () => {
        const wrapper = renderer.create(
            <Image
                id="container"
                src="http://image.com/image.png"
                alt="image"
            />
        ).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});
