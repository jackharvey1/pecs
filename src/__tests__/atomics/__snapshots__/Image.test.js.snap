// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component renders as intended 1`] = `
.c0 {
  box-shadow: 0 2px 8px 0 rgba(0,0,0,.25);
  margin: auto;
  width: 125px;
  height: 125px;
}

@media (min-width:1024px) {
  .c0 {
    width: 150px;
    height: 150px;
  }
}

<img
  alt="image"
  className="c0"
  src="http://image.com/image.png"
/>
`;
